federatedscope/cl/dataloader/Cifar10.py

```diff
@@ -0,0 +1,152 @@
+    return {
+                'client_id': {
+                    'train': DataLoader(),
+                    'test': DataLoader(),
+                    'val': DataLoader()
+                }
+            }
+    """
+    transform_train = SimCLRTransform(is_sup=False, image_size=32)
+
+    path = config.data.root
+
+    data_train = CIFAR10(path,
+                         train=True,
+                         download=True,
+                         transform=transform_train)
+    data_test = CIFAR10(path,
+                        train=False,
+                        download=True,
+                        transform=transform_train)
+
+    # Split data into dict
+    data_dict = dict()
+    data_val = data_train
+
+    data_dict = {'train': data_train, 'val': data_val, 'test': data_test}
+    data_split_tuple = (data_dict.get('train'), data_dict.get('val'),
+                        data_dict.get('test'))
+
+    config = config
+    return data_split_tuple, config
+
+
+def Cifar4LP(config):
+    r"""
+    generate Cifar10 Dataset transform and split dict for linear prob
+    evaluation of contrastive learning
+    return {
+                'client_id': {
+                    'train': DataLoader(),
+                    'test': DataLoader(),
+                    'val': DataLoader()
+                }
+            }
+    """
+    transform_train = T.Compose([
+        T.RandomResizedCrop(32,
+                            scale=(0.5, 1.0),
+                            interpolation=T.InterpolationMode.BICUBIC),
+        T.RandomHorizontalFlip(p=0.5),
+        T.ToTensor(),
+        T.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
+    ])
+    transform_test = T.Compose(
+        [T.ToTensor(),
+         T.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])
+
+    path = config.data.root
+
+    data_train = CIFAR10(path,
+                         train=True,
+                         download=True,
+                         transform=transform_train)
+    data_val = CIFAR10(path,
+                       train=True,
+                       download=True,
+                       transform=transform_test)
+    data_test = CIFAR10(path,
+                        train=False,
+                        download=True,
+                        transform=transform_test)
+
+    # Split data into dict
+    data_dict = dict()
+    data_val = data_train
+
+    data_dict = {'train': data_train, 'val': data_val, 'test': data_test}
+    data_split_tuple = (data_dict.get('train'), data_dict.get('val'),
+                        data_dict.get('test'))
+
+    config = config
+    return data_split_tuple, config
+
+
+def load_cifar_dataset(config):
+    if config.data.type == "Cifar4CL":
+        data, modified_config = Cifar4CL(config)
+        return data, modified_config
+    elif config.data.type == "Cifar4LP":
+        data, modified_config = Cifar4LP(config)
+        return data, modified_config
+
+
+register_data("Cifar4CL", load_cifar_dataset)
```

Comment:
As you use `register_data`, your function should be taken into two args:
`def load_cifar_dataset(config, client_cfgs=None):`

Reply:
I have deleted ` register_data` and finish unit test